src/analyzer_status_reporter.ts

```diff
@@ -38,8 +48,56 @@ export class AnalyzerStatusReporter extends Disposable {
 	}
 
 	private handleServerError(error: ServerErrorNotification) {
-		logError(error);
+		// Always log to the console.
+		console.error(error.message);
 		if (error.stackTrace)
 			console.error(error.stackTrace);
+
+		errorCount++;
+
+		// Offer to report the error.
+		if (config.reportAnalyzerErrors && errorCount <= maxErrorReportCount) {
+			const shouldReport: string = "Report Exception";
+			window.showErrorMessage(`Exception from the Dart analysis server: ${error.message}`, shouldReport).then(res => {
+				if (res == shouldReport)
+					this.reportError(error);
+			});
+		}
+	}
+
+	private reportError(error: ServerErrorNotification) {
+		// TODO: How to get the VSCode version?
+		let sdkVersion = getDartSdkVersion(dartSdkRoot);
+		let dartCodeVersion = extensions.getExtension('DanTup.dart-code').packageJSON.version;
+
+		let data = `
+Please report the following to https://github.com/dart-lang/sdk/issues/new:
+
+Exception from analysis server (running from VSCode)
+
+### what happened
+
+<please describe what you were doing when this exception occurred>
+
+### versions
+
+- SDK ${sdkVersion}
+- ${env.appName}
+- Dart Code ${dartCodeVersion}
+
+### the exception
+
+${error.message} ${error.isFatal ? ' (fatal)' : ''}
+
+\`\`\`
+${error.stackTrace}
+\`\`\`
+`;
+
+		let tempPath = path.join(os.tmpdir(), 'bug.md');
```

Comment:
Yeah, good idea. It looks like `tmp` might help out here: https://www.npmjs.com/package/tmp#synchronous-file-creation-1